tensorflow_addons/optimizers/average_wrapper.py

```diff
@@ -32,10 +32,20 @@ def __init__(
         if isinstance(optimizer, str):
             optimizer = tf.keras.optimizers.get(optimizer)
 
-        if not isinstance(optimizer, tf.keras.optimizers.Optimizer):
-            raise TypeError(
-                "optimizer is not an object of tf.keras.optimizers.Optimizer"
-            )
+        if tf.__version__[:3] <= "2.8":
```

Comment:
why do we need this condition? What if we would write `(tf.keras.optimizers.Optimizer, BASE_OPTIMIZER_CLASS)`?

Reply:
I did this for the error message to be accurate. but rethinking about it, I am doing (tf.keras.optimizers.Optimizer, BASE_OPTIMIZER_CLASS) check and imply that after 2.9 you should expect tf.keras.optimizers.legacy.Optimizer.